e2e/include/idata.h

```diff
@@ -23,84 +25,107 @@
 // Individual data used to store expected values from astarte
 typedef struct
 {
-    // the buffer will be freed by the idata free function
     const char *path;
-    // the buffer will be freed by the idata free function
     astarte_data_t data;
     e2e_timestamp_option_t timestamp;
-} e2e_individual_data_t;
+} idata_individual_t;
 
 // Property data used to store expected values from astarte
 // if unset is true the data element will contain an invalid value
 typedef struct
 {
-    // the buffer will be freed by the idata free function
     const char *path;
-    // the buffer will be freed by the idata free function
     astarte_data_t data;
     bool unset;
-} e2e_property_data_t;
+} idata_property_t;
 
 // Object data used to store expected values from astarte
 typedef struct
 {
-    // the buffer will be freed by the idata free function
     const char *path;
-    // the buffer will be freed by the idata free function
     e2e_object_entry_array_t entries;
-    // the buffer will be freed by the idata free function
     e2e_byte_array object_bytes;
     e2e_timestamp_option_t timestamp;
```

Comment:
Maybe is better to move all `e2e types` to this file and renaming them to `idata`, for example:
```suggestion
    idata_byte_array object_bytes
    idata_timestamp_option_t timestamp;
```